aws/resource_aws_lb_target_group.go

```diff
@@ -410,15 +410,15 @@ func validateAwsAlbTargetGroupHealthCheckProtocol(v interface{}, k string) (ws [
 	return
 }
 
-func validateAwsAlbTargetGroupPort(v interface{}, k string) (ws []string, errors []error) {
+func validateAwsLbTargetGroupPort(v interface{}, k string) (ws []string, errors []error) {
 	port := v.(int)
 	if port < 1 || port > 65536 {
 		errors = append(errors, fmt.Errorf("%q must be a valid port number (1-65536)", k))
 	}
 	return
 }
 
-func validateAwsAlbTargetGroupProtocol(v interface{}, k string) (ws []string, errors []error) {
+func validateAwsLbTargetGroupProtocol(v interface{}, k string) (ws []string, errors []error) {
 	protocol := strings.ToLower(v.(string))
 	if protocol == "http" || protocol == "https" {
```

Comment:
this needs to allow "tcp", just like your change to `validateAwsLbListenerProtocol`. "tcp" is a valid network load balancer target group protocol.